homeassistant/components/vacuum/neato.py

```diff
@@ -137,6 +144,20 @@ def update(self):
 
         self._battery_level = self._state['details']['charge']
 
+        if self._robot_has_map:
+            self._robot_map_id = self._robot_maps[self._robot_serial][0]['id']
```

Comment:
This makes no sense. We're adding things to the state that are not part of the state of the device.

We should add a websocket call instead and the frontend should show the info somewhere.

Reply:
Thank you for your review @balloob I was not sure if this was desired or not.  Do you happen to have an example of a component that uses a websocket call to display things in the front end so I can work on that?  The reason why I wanted to add the zones to the attributes is so the user can template the zone ID and they won't need to remember the ID if it changes.